src/_pytest/cacheprovider.py

```diff
@@ -382,7 +382,7 @@ def pytest_collection_modifyitems(
             self.cached_nodeids.update(item.nodeid for item in items)
 
     def _get_increasing_order(self, items: Iterable[nodes.Item]) -> List[nodes.Item]:
-        return sorted(items, key=lambda item: item.fspath.mtime(), reverse=True)
+        return sorted(items, key=lambda item: item.fspath.mtime(), reverse=True)  # type: ignore[no-any-return]
```

Comment:
I presume this ignore is because Mypy doesn't know either that `item.fspath` is an `Optional[py.path.local]`, or that `py.path.local.mtime()` returns a number.  Probably because the `Optional` case isn't handled here, which we might want to fix?

Reply:
Right now `item.fspath` is typed `Union[py.path, Any]` due to some `getattr` when it's assigned. Strictly it should be `Optional[py.path]`, because there are some nodes in which it can be `None`. But then there are a lot of places which assume it isn't (which can indeed be assumed for most nodes), and I am too lazy to sort this out just yet :)

Would be a good PR if someone wants to try it.

Reply:
Can you open an issue for that, so people know it's a PR they could make?